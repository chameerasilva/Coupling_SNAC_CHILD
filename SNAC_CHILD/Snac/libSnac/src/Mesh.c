/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
**
** Copyright (C), 2003, 
**	Steve Quenette, 110 Victoria Street, Melbourne, Victoria, 3053, Australia.
**	Californian Institute of Technology, 1200 East California Boulevard, Pasadena, California, 91125, USA.
**	University of Texas, 1 University Station, Austin, Texas, 78712, USA.
**
** Authors:
**	Stevan M. Quenette, Senior Software Engineer, VPAC. (steve@vpac.org)
**	Stevan M. Quenette, Visitor in Geophysics, Caltech.
**	Luc Lavier, Research Scientist, The University of Texas. (luc@utig.ug.utexas.edu)
**	Luc Lavier, Research Scientist, Caltech.
**
** This program is free software; you can redistribute it and/or modify it
** under the terms of the GNU General Public License as published by the
** Free Software Foundation; either version 2, or (at your option) any
** later version.
** 
** This program is distributed in the hope that it will be useful,
** but WITHOUT ANY WARRANTY; without even the implied warranty of
** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
** GNU General Public License for more details.
** 
** You should have received a copy of the GNU General Public License
** along with this program; if not, write to the Free Software
** Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
**
** $Id: Mesh.c 3274 2007-03-27 20:25:29Z EunseoChoi $
**
**~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/

#include <mpi.h>
#include <StGermain/StGermain.h>
#include <StGermain/FD/FD.h>

#include "types.h"
#include "units.h"
#include "Material.h"
#include "Context.h"
#include "Mesh.h"
#include <stdio.h>
#include <string.h>


void Snac_Mesh_Delete( void* mesh ) {
	Snac_Mesh*	self = (Snac_Mesh*)mesh;
	
	if( self->force ) {
		Memory_Free( self->force );
		self->force = 0;
	}
}

void Snac_Mesh_Print( void* mesh, Stream* stream ) {
	Snac_Mesh*	self = (Snac_Mesh*)mesh;
	Node_LocalIndex	node_lI;
	
	Journal_Printf( stream, "Snac_Mesh:\n" );
	
	Journal_Printf( stream, "\tnodeLocalCount: %u\n", self->nodeLocalCount );
}

void Snac_Mesh_Build( void* mesh, void* _context ) {
	Snac_Mesh*	self = (Snac_Mesh*)mesh;
	Snac_Context*	context = (Snac_Context*)_context;
	
	self->debug = context->debug;
	self->rank = context->rank;
	Journal_Printf( self->debug, "In: %s\n", __func__ );
	
	self->nodeLocalCount = context->mesh->nodeLocalCount;
}

void Snac_Mesh_IC( void* mesh ) {
	Snac_Mesh*	self = (Snac_Mesh*)mesh;
	
	Journal_Printf( self->debug, "In: %s\n", __func__ );
}
