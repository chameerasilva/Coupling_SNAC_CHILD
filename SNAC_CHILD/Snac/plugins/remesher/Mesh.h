/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
**
** Copyright (C), 2003,
**	Steve Quenette, 110 Victoria Street, Melbourne, Victoria, 3053, Australia.
**	Californian Institute of Technology, 1200 East California Boulevard, Pasadena, California, 91125, USA.
**	University of Texas, 1 University Station, Austin, Texas, 78712, USA.
**
** Authors:
**	Stevan M. Quenette, Senior Software Engineer, VPAC. (steve@vpac.org)
**	Stevan M. Quenette, Visitor in Geophysics, Caltech.
**	Luc Lavier, Research Scientist, The University of Texas. (luc@utig.ug.utexas.edu)
**	Luc Lavier, Research Scientist, Caltech.
**
** This program is free software; you can redistribute it and/or modify it
** under the terms of the GNU General Public License as published by the
** Free Software Foundation; either version 2, or (at your option) any
** later version.
**
** This program is distributed in the hope that it will be useful,
** but WITHOUT ANY WARRANTY; without even the implied warranty of
** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
** GNU General Public License for more details.
**
** You should have received a copy of the GNU General Public License
** along with this program; if not, write to the Free Software
** Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
**
*/
/** \file
** Role:
**	Snac exchanger properties extensions to the mesh construct.
**
** Assumptions:
**
** Comments:
**
** $Id: Mesh.h 3173 2005-11-21 23:47:09Z LukeHodkinson $
**
**~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/

#ifndef __SnacRemesher_Mesh_h__
#define __SnacRemesher_Mesh_h__


	typedef enum {
		SnacRemesher_Spherical, 
		SnacRemesher_Cartesian, 
		SnacRemesher_MeshTypeMax
	} SnacRemesher_MeshType;
	
	
	/* Mesh Information */
	struct _SnacRemesher_Mesh {
		SnacRemesher_MeshType	meshType;
		
		/* A placeholder for new coords, and stores static node coords. */
		Coord*		newNodeCoords;
		Snac_Node*	newNodes;
		Snac_Element*	newElements;
		
		/* If during the remesh nodes are found outside the old mesh they will be stored here. */
		unsigned			nExternalNodes;
		Node_LocalIndex*	externalNodes;
		unsigned			nExternalElements;
		Element_LocalIndex*	externalElements;
		
		/* Used for interpolating top/bottom internal nodes. */
		unsigned			nTopInternNodes;
		Node_LocalIndex*	topInternToLocal;
		unsigned			nBotInternNodes;
		Node_LocalIndex*	botInternToLocal;
		
		unsigned			nTopTriNodes;
		Node_DomainIndex*	topTriToDomain;
		unsigned			nBotTriNodes;
		Node_DomainIndex*	botTriToDomain;
		
		/* Used for interpolating bulk nodes. */
		unsigned			nYLines;
		unsigned*			yLineLTerm;
		unsigned*			yLineUTerm;
		unsigned*			yLineNYInds;
		unsigned**		yLineYInds;
		Node_LocalIndex**	yLineLocalInds;
		
		/* The sync class provides remote terminals for interpolating bulk nodes. */
		SnacSync*	sync;
	};
	

#if 0
	/* Mesh Information */
	struct _SnacRemesher_Mesh {
		
		Coord*		initialNodeCoord;	/* Array of initial node coordinates */
		Coord*		newNodeCoord;	/* Array of new/temporary node coordinates */
		Snac_Node*	newNode;	/* Array of new/temporary node values */
		Snac_Element*	newElement;	/* Array of new Elements */

		/* Node count... may not otherwise be directly available from this structure, so we keep a copy here */
		Node_LocalIndex		nodeLocalCount;
		Node_DomainIndex		nodeDomainCount;
		Node_GlobalIndex		nodeGlobalCount;
		Element_LocalIndex		elementLocalCount;
		Element_DomainIndex		elementDomainCount;
		Element_GlobalIndex		elementGlobalCount;

		/* Mesh type */
		SnacRemesher_MeshType		meshType;

		/* Rank neighbour info */
		Partition_Index	neighbourRankCount;
		Partition_Index*	neighbourRank;

		/* The walls (except for positive radial dir) remesh back to initial positions... maintain as one local set */
		IndexSet*		wallSet;
		Index 		wallCount;
		Index*		wallArray; /* content is local indices */

		/* The positive radial dir wall internal nodes are free... maintain as one local set */
		IndexSet*		topInternalSet;
		Index 		topInternalCount;
		Index*		topInternalArray;

		/* The positive radial dir wall internal nodes are free... maintain as one local set */
		IndexSet*		bottomInternalSet;
		Index 		bottomInternalCount;
		Index*		bottomInternalArray;

		/* Internal nodes are interpolated in the radial dir... maintain as one set */
		IndexSet*		internalSet;
		Index 		internalCount;
		Index*		internalArray; /* content is local indices */

		/* Shadow node tables */
		Node_LocalIndex*		shadowNodeLocalCount;
		Node_LocalIndex**		shadowNodeLocalArray;
		Node_ShadowIndex*		shadowNodeRemoteCount;
		Node_LocalIndex**		shadowNodeRemoteArray;
		Coord**				shadowNodeRemoteCoord;
		Snac_Node**			shadowNodeRemote;
		Element_LocalIndex*		shadowElementLocalCount;
		Element_LocalIndex**	shadowElementLocalArray;
		Element_ShadowIndex*	shadowElementRemoteCount;
		Element_GlobalIndex**	shadowElementRemoteArray;
		Snac_Element**			shadowElementRemote;
	};
#endif

	/* Print the contents of a mesh */
	void SnacRemesher_Mesh_Print( void* mesh, Stream* stream );
	

#endif /* __SnacRemesher_Mesh_h__ */

