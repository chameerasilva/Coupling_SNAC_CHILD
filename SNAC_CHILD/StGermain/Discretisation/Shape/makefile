
#Finds the Absolute path to the Project Root directory
SHELL := /bin/bash
PROJ_ROOT := $(shell until test -r ./Makefile.system ; do cd .. ; done ; echo `pwd`)
include ${PROJ_ROOT}/Makefile.system

# Subdirectories
subdirs := src tests

include ${PROJ_ROOT}/Makefile.vmake
