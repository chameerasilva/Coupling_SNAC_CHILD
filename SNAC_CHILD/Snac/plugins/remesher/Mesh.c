/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
**
** Copyright (C), 2003,
**	Steve Quenette, 110 Victoria Street, Melbourne, Victoria, 3053, Australia.
**	Californian Institute of Technology, 1200 East California Boulevard, Pasadena, California, 91125, USA.
**	University of Texas, 1 University Station, Austin, Texas, 78712, USA.
**
** Authors:
**	Stevan M. Quenette, Senior Software Engineer, VPAC. (steve@vpac.org)
**	Stevan M. Quenette, Visitor in Geophysics, Caltech.
**	Luc Lavier, Research Scientist, The University of Texas. (luc@utig.ug.utexas.edu)
**	Luc Lavier, Research Scientist, Caltech.
**
** This program is free software; you can redistribute it and/or modify it
** under the terms of the GNU General Public License as published by the
** Free Software Foundation; either version 2, or (at your option) any
** later version.
**
** This program is distributed in the hope that it will be useful,
** but WITHOUT ANY WARRANTY; without even the implied warranty of
** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
** GNU General Public License for more details.
**
** You should have received a copy of the GNU General Public License
** along with this program; if not, write to the Free Software
** Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
**
** $Id: Mesh.c 3095 2005-07-13 09:50:46Z LukeHodkinson $
**
**~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/

#include <mpi.h>
#include <StGermain/StGermain.h>
#include <StGermain/FD/FD.h>

#include "Snac/Snac.h"
#include "types.h"
#include "Mesh.h"

#include <stdio.h>


/*
** Due to the rewrite this func is severely useless.  Need to fix it.
*/

void SnacRemesher_Mesh_Print( void* mesh, Stream* stream ) {
	SnacRemesher_Mesh*	self = (SnacRemesher_Mesh*)mesh;
/*	Node_LocalIndex		node_lI; */
/*	Node_LocalIndex		index; */

	Journal_Printf( stream, "SnacRemesher_Mesh:\n" );

#if 0
	Journal_Printf( stream, "\tnodeLocalCount: %u\n", self->nodeLocalCount );
	Journal_Printf( stream, "\tnodeDomainCount: %u\n", self->nodeDomainCount );
	Journal_Printf( stream, "\tnodeGlobalCount: %u\n", self->nodeGlobalCount );
	Journal_Printf( stream, "\telementLocalCount: %u\n", self->elementLocalCount );
	Journal_Printf( stream, "\telementDomainCount: %u\n", self->elementDomainCount );
	Journal_Printf( stream, "\telementGlobalCount: %u\n", self->elementGlobalCount );
	Journal_Printf( stream, "\tinitialNodeCoord[0-%u]: { ", self->nodeLocalCount );
	for( node_lI = 0; node_lI < self->nodeLocalCount; node_lI++ ) {
		Journal_Printf( stream,
			"{ %g, %g, %g } ",
			self->initialNodeCoord[node_lI][0],
			self->initialNodeCoord[node_lI][1],
			self->initialNodeCoord[node_lI][2] );
	}
	Journal_Printf( stream, "}\n" );
	Journal_Printf( stream, "\tnewNodeCoord[0-%u]: { ", self->nodeLocalCount );
	for( node_lI = 0; node_lI < self->nodeLocalCount; node_lI++ ) {
		Journal_Printf( stream,
			"{ %g, %g, %g } ",
			self->newNodeCoord[node_lI][0],
			self->newNodeCoord[node_lI][1],
			self->newNodeCoord[node_lI][2] );
	}
	Journal_Printf( stream, "}\n" );
	Journal_Printf( stream, "\tnewNode[0-%u]: { ... TODO: Print nodes nicely... }\n", self->nodeLocalCount );

	Journal_Printf( stream,
		"\tmeshType: %s\n",
		self->meshType == SnacRemesher_Spherical ? "Spherical" :
			self->meshType == SnacRemesher_Cartesian ? "Cartesian" : "Invalid value!" );

	Journal_Printf( stream, "\tneighbourRankCount: %u\n", self->neighbourRankCount );
	Journal_Printf( stream, "\tneighbourRank[0-%u]: {", self->neighbourRankCount );
	for( index = 0; index < self->neighbourRankCount; index++ ) {
		Journal_Printf( stream, "%i ", self->neighbourRank[index] );
	}
	Journal_Printf( stream, "}\n" );

	Journal_Printf( stream, "\ttopInternalSet: " ); Print( self->topInternalSet, stream );
	Journal_Printf( stream, "\tbottomInternalSet: " ); Print( self->bottomInternalSet, stream );
	Journal_Printf( stream, "\twallSet: " ); Print( self->wallSet, stream );
	Journal_Printf( stream, "\tinternalSet: " ); Print( self->internalSet, stream );

	Journal_Printf( stream, "\ttopInternalCount: %u\n", self->topInternalCount );
	Journal_Printf( stream, "\tbottomInternalCount: %u\n", self->bottomInternalCount );
	Journal_Printf( stream, "\twalllCount: %u\n", self->wallCount );
	Journal_Printf( stream, "\tinternalCount: %u\n", self->internalCount );

	Journal_Printf( stream, "\ttopInternalArray[0-%u]: { ", self->topInternalCount );
	for( index = 0; index < self->topInternalCount; index++ ) {
		Journal_Printf( stream, "%u ", self->topInternalArray[index] );
	}
	Journal_Printf( stream, "}\n" );
	Journal_Printf( stream, "\tbottomInternalArray[0-%u]: { ", self->bottomInternalCount );
	for( index = 0; index < self->bottomInternalCount; index++ ) {
		Journal_Printf( stream, "%u ", self->bottomInternalArray[index] );
	}
	Journal_Printf( stream, "}\n" );
	Journal_Printf( stream, "\twallSet[0-%u]: { ", self->wallCount );
	for( index = 0; index < self->wallCount; index++ ) {
		Journal_Printf( stream, "%u ", self->wallArray[index] );
	}
	Journal_Printf( stream, "}\n" );
	Journal_Printf( stream, "\tinternalArray[0-%u]: { ", self->internalCount );
	for( index = 0; index < self->internalCount; index++ ) {
		Journal_Printf( stream, "%u ", self->internalArray[index] );
	}
	Journal_Printf( stream, "}\n" );

	/* Shadow */
	Journal_Printf( stream, "\tshadowNodeLocalCount[0-%u]: { ", self->neighbourRankCount );
	for( index = 0; index < self->neighbourRankCount; index++ ) {
		Journal_Printf( stream, "%u ", self->shadowNodeLocalCount[index] );
	}
	Journal_Printf( stream, "}\n" );
	Journal_Printf( stream, "\tshadowNodeLocalArray[0-%u]: {\n", self->neighbourRankCount );
	for( index = 0; index < self->neighbourRankCount; index++ ) {
		Index					index2;

		Journal_Printf( stream, "\t\tshadowNodeLocalArray[%u][0-%u]: { ", index, self->shadowNodeLocalCount[index] );
		for( index2 = 0; index2 < self->shadowNodeLocalCount[index]; index2++ ) {
			Journal_Printf( stream, "%u ", self->shadowNodeLocalArray[index][index2] );
		}
		Journal_Printf( stream, "}\n" );
	}
	Journal_Printf( stream, "}\n" );
	Journal_Printf( stream, "\tshadowNodeRemoteCount[0-%u]: { ", self->neighbourRankCount );
	for( index = 0; index < self->neighbourRankCount; index++ ) {
		Journal_Printf( stream, "%u ", self->shadowNodeRemoteCount[index] );
	}
	Journal_Printf( stream, "}\n" );
	Journal_Printf( stream, "\tshadowNodeRemoteArray[0-%u]: {\n", self->neighbourRankCount );
	for( index = 0; index < self->neighbourRankCount; index++ ) {
		Index					index2;

		Journal_Printf( stream, "\t\tshadowNodeRemoteArray[%u][0-%u]: { ", index, self->shadowNodeRemoteCount[index] );
		for( index2 = 0; index2 < self->shadowNodeRemoteCount[index]; index2++ ) {
			Journal_Printf( stream, "%u ", self->shadowNodeRemoteArray[index][index2] );
		}
		Journal_Printf( stream, "}\n" );
	}
	Journal_Printf( stream, "}\n" );
	Journal_Printf( stream, "\tshadowNodeRemoteCoord[0-%u]: {\n", self->neighbourRankCount );
	for( index = 0; index < self->neighbourRankCount; index++ ) {
		Index					index2;

		Journal_Printf( stream, "\t\tshadowNodeRemoteCoord[%u][0-%u]: { ", index, self->shadowNodeRemoteCount[index] );
		for( index2 = 0; index2 < self->shadowNodeRemoteCount[index]; index2++ ) {
			Journal_Printf( stream,
				"{ %g, %g, %g }, ",
				self->shadowNodeRemoteCoord[index][index2][0],
				self->shadowNodeRemoteCoord[index][index2][1],
				self->shadowNodeRemoteCoord[index][index2][2] );
		}
		Journal_Printf( stream, "}\n" );
	}
	Journal_Printf( stream, "}\n" );
	Journal_Printf( stream, "\tshadowNodeRemote[0-%u]: {\n", self->neighbourRankCount );
	for( index = 0; index < self->neighbourRankCount; index++ ) {
		Index					index2;

		Journal_Printf( stream, "\t\tshadowNodeRemote[%u][0-%u]: { ", index, self->shadowNodeRemoteCount[index] );
		for( index2 = 0; index2 < self->shadowNodeRemoteCount[index]; index2++ ) {
			Snac_Node_Print( &self->shadowNodeRemote[index][index2] );
		}
		Journal_Printf( stream, "}\n" );
	}
	Journal_Printf( stream, "}\n" );

	/* Shadow Elements*/
	Journal_Printf( stream, "\tshadowElementLocalCount[0-%u]: { ", self->neighbourRankCount );
	for( index = 0; index < self->neighbourRankCount; index++ ) {
		Journal_Printf( stream, "%u ", self->shadowElementLocalCount[index] );
	}
	Journal_Printf( stream, "}\n" );
	Journal_Printf( stream, "\tshadowElementLocalArray[0-%u]: {\n", self->neighbourRankCount );
	for( index = 0; index < self->neighbourRankCount; index++ ) {
		Index					index2;

		Journal_Printf( stream, "\t\tshadowElementLocalArray[%u][0-%u]: { ", index, self->shadowElementLocalCount[index] );
		for( index2 = 0; index2 < self->shadowElementLocalCount[index]; index2++ ) {
			Journal_Printf( stream, "%u ", self->shadowElementLocalArray[index][index2] );
		}
		Journal_Printf( stream, "}\n" );
	}
	Journal_Printf( stream, "}\n" );
	Journal_Printf( stream, "\tshadowElementRemoteCount[0-%u]: { ", self->neighbourRankCount );
	for( index = 0; index < self->neighbourRankCount; index++ ) {
		Journal_Printf( stream, "%u ", self->shadowElementRemoteCount[index] );
	}
	Journal_Printf( stream, "}\n" );
	Journal_Printf( stream, "\tshadowElementRemoteArray[0-%u]: {\n", self->neighbourRankCount );
	for( index = 0; index < self->neighbourRankCount; index++ ) {
		Index					index2;

		Journal_Printf( stream, "\t\tshadowElementRemoteArray[%u][0-%u]: { ", index, self->shadowElementRemoteCount[index] );
		for( index2 = 0; index2 < self->shadowElementRemoteCount[index]; index2++ ) {
			Journal_Printf( stream, "%u ", self->shadowElementRemoteArray[index][index2] );
		}
		Journal_Printf( stream, "}\n" );
	}
	Journal_Printf( stream, "}\n" );
	for( index = 0; index < self->neighbourRankCount; index++ ) {
		Index					index2;

		Journal_Printf( stream, "\t\tshadowElementRemote[%u][0-%u]: { ", index, self->shadowElementRemoteCount[index] );
		for( index2 = 0; index2 < self->shadowElementRemoteCount[index]; index2++ ) {
			Snac_Element_Print( &self->shadowElementRemote[index][index2], stream );
		}
		Journal_Printf( stream, "}\n" );
	}
	Journal_Printf( stream, "}\n" );
#endif
}
