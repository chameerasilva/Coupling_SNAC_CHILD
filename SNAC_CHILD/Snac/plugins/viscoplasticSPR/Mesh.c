/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
**
** Copyright (C), 2003,
**	Steve Quenette, 110 Victoria Street, Melbourne, Victoria, 3053, Australia.
**	Californian Institute of Technology, 1200 East California Boulevard, Pasadena, California, 91125, USA.
**	University of Texas, 1 University Station, Austin, Texas, 78712, USA.
**
** Authors:
**	Stevan M. Quenette, Senior Software Engineer, VPAC. (steve@vpac.org)
**	Stevan M. Quenette, Visitor in Geophysics, Caltech.
**	Luc Lavier, Research Scientist, The University of Texas. (luc@utig.ug.utexas.edu)
**	Luc Lavier, Research Scientist, Caltech.
**
** This program is free software; you can redistribute it and/or modify it
** under the terms of the GNU General Public License as published by the
** Free Software Foundation; either version 2, or (at your option) any
** later version.
**
** This program is distributed in the hope that it will be useful,
** but WITHOUT ANY WARRANTY; without even the implied warranty of
** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
** GNU General Public License for more details.
**
** You should have received a copy of the GNU General Public License
** along with this program; if not, write to the Free Software
** Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
**
** $Id: Mesh.c 3095 2005-07-13 09:50:46Z LukeHodkinson $
**
**~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/

#include <mpi.h>
#include <StGermain/StGermain.h>
#include <StGermain/FD/FD.h>

#include "Snac/Snac.h"
#include "types.h"
#include "Mesh.h"

#include <stdio.h>


/*
** Due to the rewrite this func is severely useless.  Need to fix it.
*/

void SnacViscoPlastic_Mesh_Print( void* mesh, Stream* stream ) {
	SnacViscoPlastic_Mesh*	self = (SnacViscoPlastic_Mesh*)mesh;

	Journal_Printf( stream, "SnacViscoPlastic_Mesh:\n" );

}
