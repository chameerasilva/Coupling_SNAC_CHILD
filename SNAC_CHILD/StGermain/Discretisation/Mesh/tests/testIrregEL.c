/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
**
** Copyright (C), 2003, Victorian Partnership for Advanced Computing (VPAC) Ltd, 110 Victoria Street, Melbourne, 3053, Australia.
**
** Authors:
**	Stevan M. Quenette, Senior Software Engineer, VPAC. (steve@vpac.org)
**	Patrick D. Sunter, Software Engineer, VPAC. (pds@vpac.org)
**	Luke J. Hodkinson, Computational Engineer, VPAC. (lhodkins@vpac.org)
**	Siew-Ching Tan, Software Engineer, VPAC. (siew@vpac.org)
**	Alan H. Lo, Computational Engineer, VPAC. (alan@vpac.org)
**	Raquibul Hassan, Computational Engineer, VPAC. (raq@vpac.org)
**
**  This library is free software; you can redistribute it and/or
**  modify it under the terms of the GNU Lesser General Public
**  License as published by the Free Software Foundation; either
**  version 2.1 of the License, or (at your option) any later version.
**
**  This library is distributed in the hope that it will be useful,
**  but WITHOUT ANY WARRANTY; without even the implied warranty of
**  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
**  Lesser General Public License for more details.
**
**  You should have received a copy of the GNU Lesser General Public
**  License along with this library; if not, write to the Free Software
**  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
**
** $Id: testIrregEL.c 3664 2006-07-04 04:26:57Z PatrickSunter $
**
**~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/

#include <stdlib.h>
#include <stdio.h>

#include <mpi.h>
#include "Base/Base.h"

#include "Discretisation/Geometry/Geometry.h"
#include "Discretisation/Shape/Shape.h"
#include "Discretisation/Mesh/Mesh.h"


int main( int argc, char* argv[] ) {
	MPI_Comm		CommWorld;
	int			rank;
	int			procCount;
	Dictionary*		dictionary;
	Geometry*		geometry;
	Topology*		topology;
	ElementLayout*		eLayout;
	Element_GlobalIndex	e_I;
	Index			i;
	XML_IO_Handler*		io_handler;
	
	/* Initialise MPI, get world info */
	MPI_Init(&argc, &argv);
	MPI_Comm_dup( MPI_COMM_WORLD, &CommWorld );
	MPI_Comm_size(MPI_COMM_WORLD, &procCount);
	MPI_Comm_rank(MPI_COMM_WORLD, &rank);

	Base_Init( &argc, &argv );
	
	DiscretisationGeometry_Init( &argc, &argv );
	DiscretisationShape_Init( &argc, &argv );
	DiscretisationMesh_Init( &argc, &argv );
	MPI_Barrier( CommWorld ); /* Ensures copyright info always come first in output */
	
	io_handler = XML_IO_Handler_New();
	dictionary = Dictionary_New();
	IO_Handler_ReadAllFromFile( io_handler, "data/trisurf.xml", dictionary );
	
	geometry = (Geometry*)IrregGeometry_New( "irregGeometry", dictionary, "geometry" );
	topology = (Topology*)IrregTopology_New( "IrregTopology", dictionary, "topology" );
	eLayout = (ElementLayout*)IrregEL_New( "IrregEL", dictionary, geometry, topology, "elements" );
	
	/*
	ElementLayout_Build( eLayout, meshDecomp );
	*/
	
	printf( "Element corner indices:\n" );
	for( e_I = 0; e_I < eLayout->elementCount; e_I++ ) {
		Index* corners = Memory_Alloc_Array( Index, eLayout->elementCornerCount, "corners" );
		
		eLayout->buildCornerIndices( eLayout, e_I, corners );
		
		printf( "\tElement %u : { %u", e_I, corners[0] );
		for( i = 1; i < eLayout->elementCornerCount; i++ )
			printf( ", %u", corners[i] );
		printf( " }\n" );
	}
	printf( "\n" );
	
	printf( "Corner element indices:\n" );
	for( i = 0; i < eLayout->cornerCount; i++ ) {
		Element_GlobalIndex	elementCnt = eLayout->cornerElementCount( eLayout, i );
		Element_GlobalIndex*	elements = Memory_Alloc_Array( Element_GlobalIndex, elementCnt, "elements" );
		
		eLayout->buildCornerElements( eLayout, i, elements );
		
		printf( "\tCorner %u : { %u", i, elements[0] );
		for( e_I = 1; e_I < elementCnt; e_I++ )
			printf( ", %u", elements[e_I] );
		printf( " }\n" );
	}
	printf( "\n" );
	
	Stg_Class_Delete( dictionary );
	Stg_Class_Delete( eLayout );
	Stg_Class_Delete( geometry );
	Stg_Class_Delete( topology );
	
	DiscretisationMesh_Finalise();
	DiscretisationShape_Finalise();
	DiscretisationGeometry_Finalise();
	
	Base_Finalise();
	
	/* Close off MPI */
	MPI_Finalize();
	
	return 0;
}
