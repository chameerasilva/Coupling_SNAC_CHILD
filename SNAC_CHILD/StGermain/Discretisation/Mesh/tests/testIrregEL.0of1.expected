StGermain Framework. Copyright (C) 2003-2005 VPAC.
Element corner indices:
	Element 0 : { 0, 1, 3 }
	Element 1 : { 1, 4, 3 }
	Element 2 : { 1, 2, 4 }
	Element 3 : { 2, 5, 4 }

Corner element indices:
	Corner 0 : { 0 }
	Corner 1 : { 0, 1, 2 }
	Corner 2 : { 2, 3 }
	Corner 3 : { 0, 1 }
	Corner 4 : { 1, 2, 3 }
	Corner 5 : { 3 }

